### PR TITLE
newsroom redux

### PR DESCRIPTION
**Pull request by nickreynolds**
- Adding newsroom data to redux so we can easily use it when rendering listings
- Might not be kicking off the API call to get newsroom data correctly

### CODE REVIEW

File: packages/dapp/src/helpers/listingEvents.ts [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
+import { getTCR, getCivil } from "./civilInstance";
 import { addListing } from "../actionCreators/listings";
+import { addNewsroom } from "../actionCreators/newsrooms";
+import { EthAddress } from "@joincivil/core";
 
 export async function initializeSubscriptions(dispatch: Dispatch): Promise<void> {
   const tcr = getTCR();
   await tcr.allApplicationsEver().forEach(async listing => {
     dispatch(addListing(listing));
+    await getNewsroom(dispatch, listing.address);
```
Comment (walfly) [inline packages/dapp/src/helpers/listingEvents.ts]: the timing of these may be bad in some situations, I would do both dispatches after the promise.

**Comment by walfly** [comment]
If you move the dispatches for listing and newsroom to after getnewsroom or make sure the newsroom is defaulted when it hasn't hydrated yet, then I"m good.

File: packages/dapp/src/components/ListingList.tsx [hunk 2]
```diff
@@ -24,10 +24,9 @@ class ListingList extends React.Component<ListingListProps> {
     return (
       <StyledUl>
         {this.props.listings.map(l => {
-          const listing = "/listing/" + l;
           return (
             <li key={l}>
-              <Link to={listing}>{l}</Link>
+              <ListingListItem listingAddress={l!} />
```
[the remark this answers is not in the document]
Comment (nickreynolds) [reply packages/dapp/src/components/ListingList.tsx]: ts cries if you don't `!`

File: packages/dapp/src/components/listing/Listing.tsx [hunk 3]
```diff
@@ -40,8 +47,9 @@ class ListingPage extends React.Component<ListingReduxProps & DispatchProp<any>
 }
 
 const mapToStateToProps = (state: State, ownProps: ListingPageProps): ListingReduxProps => {
-  const { listings, user } = state;
+  const { newsrooms, listings, user } = state;
   return {
+    newsroom: newsrooms.get(ownProps.match.params.listing),
```
[the remark this answers is not in the document]
Comment (nickreynolds) [reply packages/dapp/src/components/listing/Listing.tsx]: nevermind, it does break on refresh. fixing it a slightly different way.

File: packages/dapp/src/helpers/listingEvents.ts (hunk 1 above)
Comment (nickreynolds) [reply packages/dapp/src/helpers/listingEvents.ts]: just moving `await getNewsroom...` above `dispatch(addListing...` accomplishes that, right?

Comment (walfly) [reply packages/dapp/src/helpers/listingEvents.ts]: I think so, test it out

File: packages/dapp/src/components/listing/Listing.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (walfly) [reply packages/dapp/src/components/listing/Listing.tsx]: listing is blocked by `appExists`

**Pull request merged**